### ISSUE TITLE
crash on clicking exit button

### ISSUE THREAD

**Issue by krispyking24** [killpowa/Twickt-Launcher#1414]
**Describe the bug**
a crash when clicking the x button at the top of the window 

**To Reproduce**
1. open gd
2. Click on 'the x button in top right corrner'
3. a fail fast exception crash window

**Expected behavior**
gd launcher shouldnt crash

**Possible solutions**
Some ideas of how you would solve this issue, if possible

**Screenshots**

**Operating System:**
- OS: windows
- Version: 10
- Java Version: 8 build 1.8.0_341-b10
- Minecraft Version: 1.18.2
- Mod Loader: fabric

**Additional context**
the crash also prevents updating gd launcher so to update it i have to rerun the updated exe, heres the error that shows after it has crashed, i havent tried fully uninstall reinstalling gd yet
![image](https://user-images.githubusercontent.com/77263183/181184162-f41c6fa3-84b2-4724-8782-8b2c6583b801.png)

**Comment by Ezrahk** [comment]
I am having this issue as well, and have had it since the previous stable build as well. But I have tried a full uninstall and reinstall and that did not alleviate the problem.

OS: Windows 10 Pro
Java 8: openjdk version 1.8.0_312 64-bit
Java 17: openjdk version 17.0.1 64-bit

**Comment by Ladvace** [comment]
what version of GDLauncher do you have?

**Comment by krispyking24** [comment]
i forgot what i had at the time of the issue but rn i have the latest i think and it still happens

**Comment by Ezrahk** [comment]
Currently on version 1.1.27

**Comment by Ezrahk** [comment]
As an update, problem persists in  version 1.1.29

**Comment by Ezrahk** [comment]
Noticed a slight change in this issue and thought that I'd post about it.

When closing the main launcher, while having the Instance Manager open, AND having just closed the instance I was playing for a while does NOT cause this issue. However, closing the launcher after just launching the instance and then closing it still causes the error message.

Not sure if having the instance manager window open is a factor in the crash but thought I'd provide an update on a behavior change that I noticed.

**Comment by krispyking24** [comment]
update: it hasn't happened recently but an issue that sometimes happens is that it forgets the settings so sometimes i have to resign in and all the settings like the minecraft news dashboard at the top of gd is enabled even though i turned it off

**Comment by Eskaan** [comment]
You are not on the latest version (vl.1.29)

**Comment by krispyking24** [comment]
yes i am not rn so ill update gd but gd also never said there's an update
![image](https://user-images.githubusercontent.com/77263183/192508985-2094e557-dd13-4d4f-956d-65d36e227a43.png)

**Comment by kenalb** [comment]
I'm also having a similar error on Linux, using the Gnome desktop environment on Fedora 37.
![image](https://user-images.githubusercontent.com/109924801/219872939-775fb088-c4a2-42f3-b05e-73b6ef6fe3c8.png)
